### PR TITLE
Allow clean to remove abandoned submodules

### PR DESCRIPTION
**Pull request by Vynce**
"git clean -fdx" will not remove submodule folders that are no longer
relevant. This commit adds an additional -f argument so that unused
submodule folders are also removed.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 1]
```diff
@@ -800,7 +800,7 @@ public void submoduleClean(boolean recursive) throws GitException, InterruptedEx
     	if (recursive) {
             args.add("--recursive");
     	}
-    	args.add("git clean -fdx");
+        args.add("git clean -ffdx");
```
Comment (MarkEWaite) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: I'd rather not have the indentation change.  Can you put the white space on the front of the line back the way it was before?

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java [hunk 2]
```diff
@@ -514,7 +514,7 @@ public void execute() throws GitException, InterruptedException {
 
     public void clean() throws GitException, InterruptedException {
         reset(true);
-        launchCommand("clean", "-fdx");
+        launchCommand("clean", "-ffdx");
```
Comment (MarkEWaite) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: The last time I considered adding this, I think I found cases that warned me against it while writing tests.  

Could you provide at least one test which confirms that this truly does what you expect it to do?

**Comment by MarkEWaite** [comment]
Can you provide a similar implementation for JGit so that we keep the JGit based functionality as close to parity with the command line git implementation as we can?

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java (hunk 1 above)
Comment (Vynce) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: Sure, no problem. There's a mix of tabs and spaces used for indenting in several places, but perhaps that should be cleaned up in a separate commit.

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java (hunk 2 above)
Comment (Vynce) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: Will do.

**Comment by Vynce** [comment]
I'll see if I can figure out how to perform the same operation with JGit.

File: src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java (hunk 1 above)
Comment (MarkEWaite) [inline src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java]: I recommend against submitting a separate commit to cleanup indentation.  I'm afraid I'll just reject an indentation cleanup pull request.  That's not how this particular project chooses to operate.

Early in my submissions to the plugin I proposed that we switch the plugin to use an automatic formatting step in the maven build process. I've used automatic formatting on projects at work for years and have been very happy with the diff minimization and format standardization which results.  The other developers on the project were pretty strongly opposed to the automatic formatting idea (creates a "diff wall", damages the layout of calls to fluent interfaces, etc.)

I've accepted that, for this project, I will minimize white space changes from one commit to the next, and will ignore the inconsistencies which exist within the existing source base.

**Comment by MarkEWaite** [comment]
Please refer to the discussions in https://github.com/jenkinsci/git-client-plugin/pull/163 for more information about the impact of this type of change.

**Comment by MarkEWaite** [comment]
:-1: When a matrix project is checked out, in some cases the resulting workspace on the node which hosts the root of the matrix build will include a checkout of the repository.  Beneath that root checkout are any of the actual builds which run on that matrix build.

With that structure (root repo, with nested workspaces under the root repo), a "git clean -xffd" at the root checkout will delete the nested workspaces.

For example, I have a root of a matrix workspace where git clean -xfd reports:

```
jenkins@mark-pc1:~/jobs/JENKINS-26660-submodule-checkout-does-not-clean-deleted-submodules/workspace$ git clean -n -xfd
Would skip repository jdk/java-6/label/master
Would skip repository jdk/java-7/label/master
Would skip repository jdk/java-8/label/master
Would skip repository label/master
Would skip repository modules/firewall
Would skip repository modules/ntp
```

The same command with the "extra f" reports:

```
jenkins@mark-pc1:~/jobs/JENKINS-26660-submodule-checkout-does-not-clean-deleted-submodules/workspace$ git clean -n -xffd
Would remove jdk/
Would remove label/
Would remove modules/
```
